### DISCUSSION TITLE
Wish cards had a separate "full size" reading/editing view!

### DISCUSSION THREAD

**Discussion by michaellw** [ycnmhd/obsidian-lineage#21] [Ideas]
I wonder if it would be possible to have only the content of the current card in the right editor window, instead of the content of the whole document.

**Comment by ycnmhd** [comment]
Do you mean in a regular Obsidian markdown split-view? if not, could you make a basic wireframe of this idea using [excalidraw](https://excalidraw.com/)?

**Reply by michaellw** [reply]
Yes, my first thought was in a regular Obsidian markdown split-view, just like "Open to the right" of a file, it opens editing window on the right, however, the contents of the window are not the contents of the entire document, but only the contents of the currently selected card.

**Reply by michaellw** [reply]
Another way is that when open the right editing window with "Jump to section", there is an option or toggle button to hide the content of other cards/sections and only show the content of the current card/section.

**Reply by michaellw** [reply]
On top of the second way, I came up with another little feature request. In the right editing window opened with "Jump to section", highlight the contents of the current card/section and lowlight the contents of the other cards/sections. This one should probably be a separate one.

**Reply by ycnmhd** [reply]
Overriding and managing a separate Obsidian split-view would be too hacky/brittle.
I think a more stable and maintainable approach would be to split the plugin view itself.
![image](https://github.com/ycnmhd/obsidian-lineage/assets/33285087/b501e70b-bf21-4627-a5a6-b1744aff1757)

**Reply by michaellw** [reply]
I think you're right, the only thing I'd like is for the plugin edit view to be compatible with the obsidian standard markdown view for compatibility with other plugins renderings.

**Reply by ycnmhd** [reply]
It would be the same editor that is used in regular cards.
In some cases, compatibility with other plugins might require explicit support from their end, because the editor view is not mounted in a conventional way.

**Reply by ycnmhd** [reply]
Editing a card in a split sidebar has been implemented in 0.7.0
![image](https://github.com/user-attachments/assets/21bb43d8-c8c9-430d-9c2b-bd8c1f462885)

**Comment by FeralFlora** [comment]
Perhaps another way to implement something like this would be to make the cards into embedded Obsidian editors that you could then expand. 

This is what the [Kanban plugin 2.0](https://github.com/mgmeyers/obsidian-kanban) uses for its cards (released out of beta today! 🎉), and this is also how the comment gutter in the [CriticMarkup / Commentator plugin](https://github.com/Fevol/obsidian-criticmarkup) is implemented.

Here's what [mgmeyers wrote](https://discord.com/channels/686053708261228577/855181471643861002/1241431001172017202) about the benefits of this approach:
> Kanban cards now use Obsidian’s native markdown editor. This brings with it improved tag and link autocompletion as well as support for plugins that modify the editing experience. It also means that things like dataview queries will display in cards.

See a demo of the comment gutter in Commentator here: https://discord.com/channels/686053708261228577/1082420499163918408/1231693594461474886

Fevol, the developer of the Commentator plugin, has extracted the relevant editor embedding and view code from the Kanban plugin into a gist here:
https://gist.github.com/Fevol/caa478ce303e69eabede7b12b2323838

It has an MIT licence!

**Reply by ycnmhd** [reply]
Lineage uses an embedded Obsidian editor in [editor mode](https://github.com/ycnmhd/obsidian-lineage/blob/master/src/obsidian/helpers/inline-editor.ts#L92) , and the [MarkdownRenderer](https://docs.obsidian.md/Reference/TypeScript+API/MarkdownRenderer/render) in [preview mode](https://github.com/ycnmhd/obsidian-lineage/blob/a0a373bc091a1d42bf5703c4d699a17ac6eaf4ef/src/view/actions/markdown-preview-action.ts#L17) (for performance reasons).

If you find any (behavioral) differences between these two approaches (i.e., plugins or hotkeys that work with Fevol's approach but not with Lineage) please submit an issue.
(I just tested Dataview and Hover editor and they both work.)

**Reply by FeralFlora** [reply]
Ah, okay, thanks for clarifying. Please excuse my ignorance, I just started testing the plugin, so I wasn't aware of its full capabilities. That's great!
